lib/Basic/Default/TaskQueue.inc

```diff
@@ -100,10 +103,22 @@ bool TaskQueue::execute(TaskBeganCallback Began, TaskFinishedCallback Finished,
         T->Env.empty() ? decltype(Envp)(None)
                        : decltype(Envp)(llvm::toStringRefArray(T->Env.data()));
 
+    llvm::SmallString<64> stdoutPath;
+    llvm::SmallString<64> stderrPath;
+    if (fs::createTemporaryFile("stdout", "tmp",  stdoutPath)
+        || fs::createTemporaryFile("stderr", "tmp",  stderrPath)) {
+      return true;
+    }
+
+    llvm::sys::RemoveFileOnSignal(stdoutPath);
+    llvm::sys::RemoveFileOnSignal(stderrPath);
+
+    SmallVector<Optional<StringRef>, 4> redirects = {None, {stdoutPath}, {stderrPath}};
```

Comment:
Last nitpick: This repo has an 80-column limit. But you can solve that anyway by using a plain fixed-sized array instead of a SmallVector.

```cxx
Optional<StringRef> redirects[] = {None, {stdoutPath}, {stderrPath}};
```

Reply:
Fixed.